astrapy/idiomatic/collection.py

```diff
@@ -216,6 +216,9 @@ def delete_many(
                 f"(gotten '${json.dumps(dm_response)}')"
             )
 
+    def drop(self, name_or_collection: Union[str, AsyncCollection]) -> None:
```

Comment:
As a method on "collection", it has no "name_or_collection" argument (see [ref](https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html#pymongo.collection.Collection.drop)), it being implied that the one to drop is self.

note: The Mongo docs use the term "alias" pretty liberally here IMO :D